From d7ffe23551bfea7c3ead2275ef98c51e0e13a1fe Mon Sep 17 00:00:00 2001
From: Rob Colclaser <rcolclaser@marvell.com>
Date: Wed, 13 Mar 2019 15:51:14 +0530
Subject: [PATCH 039/386] octeontx2-af: Fix return value in NIX PSE errata
 workaround

Change-Id: Ia080e24e0781eb7e6d2a0d47fd73b14161ca5d58
Signed-off-by: Rob Colclaser <rcolclaser@marvell.com>
Signed-off-by: Geetha sowjanya <gakula@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/5616
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_fixes.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_fixes.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_fixes.c
index 7c89e0b94e6a..1fc2ee12f9ae 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_fixes.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_fixes.c
@@ -384,6 +384,7 @@ static bool is_sq_stalled(struct rvu *rvu, struct nix_hw *nix_hw, int smq)
 		tx_stall->nixlf_stall_count[nixlf]++;
 		return true;
 	}
+	return false;
 
 poll_sq_stats:
 	if (!tx_stall->nixlf_tl2_count[nixlf])
-- 
2.17.1

